staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_filter.go

```diff
@@ -162,6 +163,7 @@ func (cfgCtlr *configController) Handle(ctx context.Context, requestDigest Reque
 	if req == nil {
 		if queued {
 			metrics.ObserveWaitingDuration(ctx, pl.Name, fs.Name, strconv.FormatBool(req != nil), time.Since(startWaitingTime))
```

Comment:
Is it possible to modify the ObserveWaitingDuration function instead of adding this next to it every place?

Reply:
Definitely possible since the latency tracker is stored in the context, but the latency tracker is not itself a metric so I would find it unexpected if `ObserveWaitingDuration` did anything more than just recording the metric it's meant for.

Reply:
I added a `observeQueueWaitTime` helper function instead that calls both